Pinta.Core/ImageFormats/FormatDescriptor.cs

```diff
@@ -73,22 +75,24 @@ public sealed class FormatDescriptor
 		/// <param name="mimes">A list of supported file MIME types (for example, "image/jpeg" and "image/png").</param>
 		/// <param name="importer">The importer for this file format, or null if importing is not supported.</param>
 		/// <param name="exporter">The exporter for this file format, or null if exporting is not supported.</param>
-		public FormatDescriptor (string displayPrefix, string[] extensions, string[] mimes,
+		public FormatDescriptor (string displayPrefix, IEnumerable<string> extensions, IEnumerable<string> mimes,
 					 IImageImporter? importer, IImageExporter? exporter)
 		{
-			if (extensions == null || (importer == null && exporter == null)) {
+			if (extensions == null || (importer == null && exporter == null))
 				throw new ArgumentNullException ("Format descriptor is initialized incorrectly");
-			}
 
-			this.Extensions = extensions;
-			this.Mimes = mimes;
+			var materializedExtensions = extensions.ToReadOnlyCollection ();
```

Comment:
Are these separate local variables needed, or can the subsequent code just use `this.Extensions` instead?